feat: adding CLI tool on the website (449)

boyney123 opened the pull request:
**Description**

Here is a solution for #449, which adds the CLI within our tools, I believe it can help new users find the CLI and start using it.

Commit:
adding CLI tool on the website

boyney123 commented:
What do you think @fmvilas @derberg , I think the new page could help? Or maybe the CLI is already on the website that I'm missing?

@alequetzalli if this goes ahead, will need some help with the wording and stuff 🙏, I kinda put some stuff together, but no idea really if things make sense lol

derberg commented:
I'm not sure how `asyncapi new` works, but maybe it could be added there into code snippet, as a usage example, as it is with the generator?

derberg commented:
@boyney123 I think you did not provide a correct link in the description

boyney123 commented:
> @boyney123 I think you did not provide a correct link in the description

ah, thanks fixed that

boyney123 commented:
> I'm not sure how `asyncapi new` works, but maybe it could be added there into code snippet, as a usage example, as it is with the generator?

![image](https://user-images.githubusercontent.com/3268013/139882192-2929c4e6-bb65-4739-a903-79b4c91ee655.png)

Hey @derberg do you mean here on the screenshot? Not 100% sure what you mean, you want to show an example of `asyncapi new`?

derberg commented:
yes exactly, there where you pointed and yes, show example usage of `asyncapi` that anyone can copy and paste, like it is with `Generator` view in the website. I think only `asyncapi new` is feasible for such a use, as far as I remember from Fran's demo on its use case

Commit:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+  {
+    name: "New files",
+    description:
+      "Use the CLI tool to quickly create new AsyncAPI files. Select from a range of templates (MMQT, WebSockets, Kafka and more...)",
```
```suggestion
      "Use the CLI tool to quickly create new AsyncAPI files. Select from a range of templates (MMQT, WebSockets, Kafka, and more.)",
```

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+  {
+    name: "Validate",
+    description:
+      "Validate your AsyncAPI files with the CLI. Quickly get feedback to verify your AsyncAPI is within the correct format.",
```
```suggestion
      "Validate your AsyncAPI files with the CLI. Quickly get feedback to verify your AsyncAPI file is within the correct format.",
```

Would it make sense to say there `AsyncAPI document` or `AsyncAPI file`?

boyney123 replied:
Yeah interesting, I have no idea 🤣, but I do think `AsyncAPI document` sounds better.

Any idea what's preferred @derberg @fmvilas ?

boyney123 replied:
Going to add document for now 👍

derberg replied:
it is complicated, I personally always use `file` unless technically it doesn't make sense and then I say `document` 😄 but basically trying to avoid `document` as much as possible.

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+  {
+    name: "Open Source",
+    description:
+      "All our tools are open source, feel free to contribute new commands or help evolve our exisiting ones",
```
```suggestion
      "All our tools are open source, feel free to contribute new commands or help evolve our existing ones.",
```

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+  }
+
+  const description =
+    "Create, validate and explore your AsyncAPI files with our CLI tool";
```
```suggestion
    "Create, validate, and explore your AsyncAPI files with our CLI tool.";
```

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+              </h4>
+              <p className="mt-3 text-lg leading-7 text-gray-500 lg:pr-4">
+                Start using AsyncAPI CLI really quickly. Select one of our
+                commands to get started.
```
```suggestion
                Start using AsyncAPI CLI within seconds by selecting one of our
                commands to get started.
```

quetzalliwrites commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,136 @@
+              Features
+            </h4>
+            <p className="mt-3 text-lg leading-7 text-gray-500 lg:pr-4">
+              Use the AsyncAPI CLI tool to help you create, develop and maintain
```
```suggestion
              Use the AsyncAPI CLI tool to help you create, develop, and maintain
```

quetzalliwrites reviewed (commented):
Just a few fixes to do, but I really liked what you wrote 😀👍🏽✨✨✨

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

boyney123 commented:
Thanks for the review @alequetzalli 👍

Commit:
adding examples of the cli page

Commit:
Merge branch '448-adding-cli-as-tool-on-website' of github.com:boyney123/website into 448-adding-cli-as-tool-on-website

boyney123 commented:
@derberg thanks for the review and suggestion, I added some examples on the page, let me know what you think

derberg commented:
@boyney123 looks awesome, the only remark comes from my old times when I was a tech writer 😄 I think it should be `AsyncAPI Studio` not `AsyncAPI studio` and that is it from my side, once you fix it I will approve

boyney123 commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,175 @@
+                    />
+                  </div>
+                  <div>
+                    <h4 className="text-xs">Open AsyncAPI studio</h4>
```
```suggestion
                    <h4 className="text-xs">Open AsyncAPI Studio</h4>
```

boyney123 committed:
Update pages/tools/cli.js

Commit:
fixing small issues with new command changes

boyney123 commented:
thanks @derberg , all sorted 👍

derberg reviewed (dismissed):
well done 👏🏼

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

derberg commented:
@magicmatatjahu @fmvilas anything to add from your side folks?

magicmatatjahu commented:
Everything is good, except that 

![image](https://user-images.githubusercontent.com/20404945/140082052-5399efe4-7bf2-405b-b249-e4d16467fbf5.png)

Can we render it (`asyncapi studio`) as inline code like in markdown?

fmvilas commented:
It's also `asyncapi start studio` not `asyncapi studio` :)

Commit:
fixing command and way the command is rendered in the feature

boyney123 commented:
> Can we render it (asyncapi studio) as inline code like in markdown?

@magicmatatjahu yeah I tried to use the CodeBlock but it looked very odd, so I might a small change to add some background and stuff, to give the same effect, let me know you think 👍

magicmatatjahu commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,177 @@
+  {
+    name: "Open Studio",
+    // eslint-disable-next-line react/display-name
+    description: () => <>Got an AsyncAPI file locally? Run <span className="bg-gray-800 text-gray-100 rounded-sm px-2 py-0.5">asyncapi start studio</span> to open our studio in seconds.</>,
```
```suggestion
    description: () => <>Got an AsyncAPI file locally? Run <code className=" px-1 py-0.5 bg-primary-700 text-white rounded font-mono text-sm">asyncapi start studio</code> to open our studio in seconds.</>,
```

and then it will be rendered as ![image](https://user-images.githubusercontent.com/20404945/141136554-97bdde2f-eae6-43c6-bd92-9a12e0b09889.png)`

:) We should be consistent with other views

boyney123 replied:
ah cool thanks @magicmatatjahu !

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Maciej Urbańczyk <urbanczyk.maciej.95@gmail.com>

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

fmvilas commented on pages/tools/cli.js:
```diff
@@ -0,0 +1,177 @@
+  {
+    name: "New files",
+    description: () =>
+      "Use the CLI tool to quickly create new AsyncAPI files. Select from a range of templates (MMQT, WebSockets, Kafka, and more.)",
```
```suggestion
      "Use the CLI tool to quickly create new AsyncAPI files. Select from a range of templates (MQTT, WebSockets, Kafka, and more.)",
```

boyney123 committed:
Update pages/tools/cli.js

Co-authored-by: Fran Méndez <fmvilas@gmail.com>

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

magicmatatjahu committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

boyney123 commented:
Not sure if I'm missing something but GitHub telling me that there are changes requested, but I think I added your changes @fmvilas 

![image](https://user-images.githubusercontent.com/3268013/142776708-fa86bd02-619a-428e-9a5a-3d333760dba4.png)


Think this PR is ready for a new review 🙏 @fmvilas @magicmatatjahu

magicmatatjahu commented:
> Not sure if I'm missing something but GitHub telling me that there are changes requested, but I think I added your changes

If someone reject PR in the past, have to accept it after changes, reject isn't missed in this case :)

derberg commented:
I don't think we should merge before https://github.com/asyncapi/cli/pull/130 is fixed, otherwise we will promote command that is not really copy/paste. Other solution, just have `asyncapi new` and `asyncapi help`

fmvilas commented:
Yeah, agree with @derberg. One thing you can folks do is to remove the `start studio` example and leave only `asyncapi new` as it already asks you if you want to open Studio 🤷‍♂️ Or just wait for the other issue to get fixed, as you prefer.

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

boyney123 commented:
Cool thanks @derberg @fmvilas yeah OK let's get asyncapi/cli#130 fixed first 👍

derberg commented:
@boyney123 @fmvilas I know asyncapi/cli#130  is merged but to be honest it is not yet fully solved, I mean the impression that users will have when trying to run `npm start studio`. We fallback to local file, cool, but most users will not have it when running CLI basing on examples on the website, and they will get a not very nice error that is not yet solved, which is `that context is missing` instead of nice error with explanation of what options user have to provide asyncapi file.

I'm not sure it is worth blocking this PR before all these issues are solved, I suggest we just remove `asyncapi start studio` and only keep `asyncapi new` and `asyncapi help` and that is it. Thoughts?

boyney123 committed:
Merge branch 'master' into 448-adding-cli-as-tool-on-website

boyney123 commented:
@derberg thinking about removing it from the list of examples, but maybe we keep it on the section (highlighted in image below)?.

Or you think we should remove that too, and replace it? If so any ideas?

![image](https://user-images.githubusercontent.com/3268013/143573020-85adbbfb-976e-40f1-84b8-318d8806f0d4.png)

derberg commented:
@boyney123 perfect solution imho 🚀

Commit:
removed example for async studio command

boyney123 commented:
OK cool, I just pushed up the changes 👍

magicmatatjahu reviewed (changes requested):
Do we really need second link to Github repo (see screenshot)?

![image](https://user-images.githubusercontent.com/20404945/143577787-b0cbf709-8df7-479c-9c70-b2730862338e.png)

I also think that `Features` and description for it should be centered :)

Also, please update metadata for new page, because I see this, when I go to `/tools/cli` page :)

![image](https://user-images.githubusercontent.com/20404945/143577870-ec3239fd-43fa-4ec4-bad4-e1306e043318.png)

derberg commented:
@magicmatatjahu 

> Do we really need second link to Github repo

it is consistent with `Generator` view and I think it is ok, kinda makes sense

> I also think that Features and description for it should be centered 

I like it is on the right as it is consistent with other tools view, even if on others it is caused by images on the right

> Also, please update metadata for new page, because I see this, when I go to /tools/cli page :)

you sir have 🦅 👀

magicmatatjahu commented:
> it is consistent with Generator view and I think it is ok, kinda makes sense

We can always remove that button also in `Generator` view :)

> I like it is on the right as it is consistent with other tools view, even if on others it is caused by images on the right

It can stay, but yeah, without any content on right it's not a "nice" to see.

> you sir have 🦅 👀

I don't know if I should thank you, because I smell trolling 😅

derberg commented:
@magicmatatjahu 

> I don't know if I should thank you, because I smell trolling 😅

Your nose is playing tricks on you on Monday 😄 
I really do think you did greate there, nobody else spotted 🤷🏼‍♂️

Commit:
fixing title to match CLI

Commit:
fixing cli layout issues on mobile and tablet

boyney123 commented:
> I also think that Features and descriptions for it should be centered :)

Now looking at it again, yeah agree mate, thanks! Changed think it looks better. @derberg thoughts? I know you prefer the other way, but after changing I think it looks slightly better 🤔 ?

> Also, please update metadata for new page, because I see this, when I go to /tools/cli page :)

Ah, nice find! Updated!

> Do we really need second link to the Github repo

Yeah, it does look quite odd 🤔 , but as @derberg I only did it to be consistent with other pages, happy to make the changes if we want to. Maybe another PR if we have issues with it?

---

Also found a few layout issues on mobile + tablet, they have also been fixed.

magicmatatjahu reviewed (approved):
LGMT! @derberg What do you think about this centered content? Do you accept that?

derberg reviewed (approved):
lgtm 🚀

derberg commented:
I think we still need @fmvilas to accept as his previous reject is blocking merge

fmvilas reviewed (approved):
👏  :shipit:
